### PR TITLE
Add UI for dependsOn

### LINKED ISSUE #570: No UI for dependsOn
There's no UI for [dependsOn](https://yml.publiccode.tools/schema.core.html#section-dependson) which makes it impossible for users to generate a publiccode.yml with those fields or to fix one with errors.

### PR DESCRIPTION
**Pull request by ale-saglia**
## Summary

- add separate editors for open source, proprietary, and hardware dependencies
- support adding and removing dependencies and editing every `dependsOn` field
- preserve imported dependencies during YAML round trips and omit empty rows
- provide accessible labels, descriptions, focus handling, and translations for every supported UI language

## Why

The editor already modeled, imported, and serialized `dependsOn`, but it did not render controls for the field. Users therefore could not add dependencies or inspect and edit values loaded from an existing `publiccode.yml`.

## Validation

- `npm test -- --runInBand` (69 tests)
- `npm run lint`
- `npx tsc -b`
- `npm run build`
- `npx prettier --check ...`

Closes #570

### CODE REVIEW

**Comment by valeriocomo** [comment]
ciao @ale-saglia. Thank you for this contribution.

Just some consideration about UI

## 1

A message stating that there's no dependency is missing. For reference, take a look at _Funded By_ field.

image

## 2

Field description must be provided by a tooltip on a information icon and not right below the field name

image

## 3

Form UI is overall different from other lists with complex object, such as _contractor_, _contact_ or _funded by._ Using a recurring pattern would enhance perceived quality and it gives coherence to the whole UI.


I would ask you to reconsider these points and make the necessary changes.
BTW, a review of the code is ongoing.

**Comment by ale-saglia** [comment]
Hi @valeriocomo, thanks for the feedback. I addressed the three UI points in 3373d88:

1. Added an explicit **“No dependencies present”** empty state to each dependency group.
2. Moved dependency and field descriptions to accessible information-icon tooltips.
3. Reworked each group as a responsive table, following the recurring pattern used by **Funded By**, contacts, and contractors.

I also added focused tests for the empty states and tooltips while preserving the existing add/remove focus handling and YAML round-trip behavior. Locally, all 71 tests pass; lint has no errors; typecheck and build pass.

Empty states:

Depends On empty states

Table layout and field tooltip:

Depends On table layout and tooltip

File: src/app/components/EditorDependencies.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          type="button"
+        >
+          {t("editor.form.addnew")}
+        </Button>
+      </fieldset>
+    </div>
+  );
+}
+
+export default function EditorDependencies(): JSX.Element {
```
Comment (valeriocomo) [inline src/app/components/EditorDependencies.tsx]: Must be renamed to EditorDependsOn, filename and tests too.

**Review by valeriocomo — CHANGES_REQUESTED** [review]
I realize that _Reset_ action doesn't reset the _dependsOn_ form field. Could you fix this problem?

**Comment by valeriocomo** [comment]
@AlexPalaz could you review this PR as well?

File: src/app/components/EditorDependencies.tsx (hunk 1 above)
Comment (ale-saglia) [reply src/app/components/EditorDependencies.tsx]: Renamed the component, source file, and test file to `EditorDependsOn` in 9c5ef38.

**Comment by ale-saglia** [comment]
Addressed the reset issue in 9c5ef38. The editor reset defaults now provide empty arrays for all three `dependsOn` groups, which React Hook Form requires to reset each field array, and a regression test confirms that existing dependency rows are cleared.

The same commit also renames the component, source file, and test file to `EditorDependsOn`.

Validation: 72 tests pass; lint has no errors (5 pre-existing warnings); typecheck and production build pass.

**Review by AlexPalaz — APPROVED** [review]

**Review by valeriocomo — APPROVED** [review]
LGTM
